subsys/net/lib/sockets/sockets.c

```diff
@@ -2530,6 +2581,35 @@ static int sock_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 		return 0;
 	}
 
+	case ZFD_IOCTL_FIONBIO:
+		sock_set_flag(obj, SOCK_NONBLOCK, SOCK_NONBLOCK);
+		return 0;
+
+	case ZFD_IOCTL_FIONREAD: {
+		int *avail;
+		ssize_t len;
+		struct net_pkt *pkt;
+		struct net_context *ctx = (struct net_context *)obj;
+
+		avail = va_arg(args, int *);
+		pkt = k_fifo_peek_head(&ctx->recv_q);
+		if (pkt == NULL) {
+			*avail = 0;
+			return 0;
+		}
+
+		len = net_pkt_get_len(pkt);
+		/* TODO: account for other headers */
```

Comment:
We have net_pkt_ip_hdr_len() helper that should return the IP header length, perhaps we could use it here?